pupil_src/shared_modules/pupil_remote.py

```diff
@@ -67,45 +68,106 @@ class Pupil_Remote(Plugin):
+        response,msg = self.thread_pipe.recv_multipart()
+        if response == 'Bind OK' :
+            host,port = msg.split(':')
+            self.host = host
+            self.port = port
+            return
+
+        #fail logic
+        logger.error(msg)
+
+        #for service we shut down
+        if self.g_pool.app == 'service':
+            audio.say("Error: Port already in use.")
+            self.notify_all({'subject':'service_process.should_stop'})
+            return
+
+        #for capture we try to bind to a arbitrary port on the first external interface
         else:
-            logger.error(response)
-            if self.g_pool.app == 'service':
-                audio.say("Error: Port already in use.")
-                self.notify_all({'subject':'service_process.should_stop'})
+            self.thread_pipe.send_multipart(('Bind',"tcp://*:*"))
+            response,msg = self.thread_pipe.recv_multipart()
+            if response == 'Bind OK' :
+                host,port = msg.split(':')
+                self.host = host
+                self.port = port
             else:
-                self.address = ''
+                logger.error(msg)
+                raise Exception("Could not bind to port")
+
 
     def stop_server(self):
         self.thread_pipe.send('Exit')
         while self.thread_pipe:
             sleep(.1)
 
     def init_gui(self):
+        self.menu = ui.Growing_Menu('Pupil Remote')
+        self.g_pool.sidebar.append(self.menu)
+        self.update_menu()
+
+    def update_menu(self):
+
+        self.menu.elements[:] = []
+
         def close():
             self.alive = False
 
+        def set_iface(use_primary_interface):
+            self.use_primary_interface = use_primary_interface
+            self.update_menu()
+
+
+
+        if self.use_primary_interface:
+            def set_port(new_port):
+                new_address = '*:'+new_port
+                self.start_server(new_address)
+                self.update_menu()
+
+        else:
+            def set_port(new_port):
+                new_address = self.host+':'+new_port
+                self.start_server(new_address)
+                self.update_menu()
+
+
+            def set_address(new_address):
+                if new_address.count(":") != 1:
+                    logger.error("address format not correct")
+                    return
+                self.start_server(new_address)
+                self.update_menu()
+
+
         help_str = 'Pupil Remote using ZeroMQ REQ REP scheme.'
-        self.menu = ui.Growing_Menu('Pupil Remote')
         self.menu.append(ui.Button('Close',close))
         self.menu.append(ui.Info_Text(help_str))
-        self.menu.append(ui.Text_Input('address',self,setter=self.start_server,label='Address'))
-        self.g_pool.sidebar.append(self.menu)
+        self.menu.append(ui.Switch('use_primary_interface',self,setter=set_iface,label="use primary network interface"))
```

Comment:
I think "use" should be capitalized here. Generally, this is something we should have a look at.